RecoTauTag/HLTProducers/interface/L1TJetsMatching.h

```diff
@@ -0,0 +1,210 @@
+// -*- C++ -*-
+//
+// Package:    RecoTauTag/HLTProducers
+// Class:      L1TJetsMatching
+// 
+/**\class L1TJetsMatching L1TJetsMatching.h 
+ RecoTauTag/HLTProducers/interface/L1TJetsMatching.h
+ Description: 
+ Matching L1 to PF/Calo Jets. Used for HLT_VBF paths.
+	*Matches PF/Calo Jets to L1 jets from the dedicated seed
+	*Adds selection criteria to the leading/subleading jets as well as the maximum dijet mass
+	*Separates collections of PF/Calo jets into two categories
+ 
+ 
+*/
+//
+// Original Author:  Vukasin Milosevic
+//         Created:  Thu, 01 Jun 2017 17:23:00 GMT
+//
+//
+
+
+
+
+#ifndef RecoTauTag_HLTProducers_L1TJetsMatching_h
+#define RecoTauTag_HLTProducers_L1TJetsMatching_h
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/global/EDProducer.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/JetReco/interface/PFJetCollection.h"
+#include "DataFormats/TauReco/interface/PFTauFwd.h"
+#include "DataFormats/HLTReco/interface/TriggerFilterObjectWithRefs.h"
+
+#include "Math/GenVector/VectorUtil.h"
+#include "DataFormats/HLTReco/interface/TriggerTypeDefs.h"
+#include "FWCore/Utilities/interface/EDMException.h"
+#include "DataFormats/JetReco/interface/PFJet.h"
+
+#include "HLTrigger/HLTcore/interface/defaultModuleLabel.h"
+#include "DataFormats/Math/interface/deltaR.h"
+
+#include <map>
+#include <vector>
+
+template< typename T>
+class L1TJetsMatching: public edm::global::EDProducer<> {
+ public:
+  explicit L1TJetsMatching(const edm::ParameterSet&);
+  ~L1TJetsMatching();
+  virtual void produce(edm::StreamID, edm::Event&, const edm::EventSetup&) const override;
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+ private:
+    
+  const edm::EDGetTokenT<std::vector<T>> jetSrc_;
+  const edm::EDGetTokenT<trigger::TriggerFilterObjectWithRefs> jetTrigger_;
+  const double pt1Min;
+  const double pt2Min;
+  const double mjjMin;
+  const double matchingR;
+  const double matchingR2;
```

Comment:
Sorry, it the trailing underscore text showed up in the email, but the pt1Min(_) didn't. I'll change it